modules/jooby-avaje-inject/src/main/java/io/jooby/avaje/inject/AvajeInjectModule.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Jooby https://jooby.io
+ * Apache License Version 2.0 https://jooby.io/LICENSE.txt
+ * Copyright 2014 Edgar Espina
+ */
+package io.jooby.avaje.inject;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.typesafe.config.Config;
+
+import io.avaje.inject.BeanScope;
+import io.avaje.inject.BeanScopeBuilder;
+import io.jooby.Extension;
+import io.jooby.Jooby;
+
+/**
+ * Avaje Inject module: https://jooby.io/modules/avaje-inject.
+ *
+ * <p>Jooby integrates the {@link io.jooby.ServiceRegistry} into the Avaje DI framework.
+ *
+ * <p>Usage:
+ *
+ * <pre>{@code
+ * {
+ *
+ *
+ *   install(new AvajeInjectModule());
+ *
+ * }
+ *
+ * }</pre>
+ *
+ * Require calls are going to be resolved by Avaje inject now.
+ *
+ * @author josiah
+ * @since 3.0.0
+ */
+public class AvajeInjectModule implements Extension {
+
+  private final BeanScopeBuilder beanScope;
+
+  public static AvajeInjectModule of() {
+    return new AvajeInjectModule(BeanScope.builder());
+  }
+
+  public static AvajeInjectModule of(BeanScopeBuilder beanScope) {
+    return new AvajeInjectModule(beanScope);
+  }
+
+  AvajeInjectModule(BeanScopeBuilder beanScope) {
+    this.beanScope = beanScope;
+  }
+
+  @Override
+  public void install(Jooby application) throws Exception {
+
+    application
```

Comment:
This probably better to run on `starting` or `started`. So we don't depend of sequential order of services being added before this module.

Reply:
would lateInit=true resolve?

Reply:
Tested, with lateinit=true the order does not matter anymore.

Reply:
right. lateinit will fix it